src/System.Memory/src/System/Buffers/ArrayMemoryPool.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Buffers
+{
+    internal sealed partial class ArrayMemoryPool<T> : MemoryPool<T>
+    {
+        private const int DefaultSize = 4096;
+
+        private static readonly ArrayMemoryPool<T> s_shared = new ArrayMemoryPool<T>();
```

Comment:
This is unnecessary extra layer `MemoryPool<T> Shared` can allocate ArrayMemoryPool<T> directly.

Reply:
Yeah, I think we should use the shared pool.